spec/rubocop/cop/infinum/factory_bot_association_spec.rb

```diff
@@ -0,0 +1,79 @@
+RSpec.describe RuboCop::Cop::Infinum::FactoryBotAssociation, :config do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+  let(:message) { "Use #{association_name} { build(:#{factory_name}) } instead" }
+
+  context 'when association name is foo and factory name is foo' do
+    let(:association_name) { 'foo' }
+    let(:factory_name) { 'foo' }
+
+    it 'disallows method' do
+      expect_offense(<<~RUBY)
+        association :foo
+        ^^^^^^^^^^^^^^^^ #{message}
+      RUBY
+    end
+
+    it 'disallows inline method' do
+      expect_offense(<<~RUBY)
+        foo { association :foo }
+              ^^^^^^^^^^^^^^^^ #{message}
```

Comment:
So, if I see correctly, this would mean that the warning will look like so:

```
foo { association :foo }
         ^^^^^^^^^^^ Use foo { build(:foo) } instead
```
which gives the impression that the cop wants:
```
foo { foo { build(:foo) } }
```

In other words, underline the whole thing if this happens:
```
foo { association :foo }
^^^^^^^^^^^^^^^^ Use foo { build(:foo) } instead
```

Reply:
Weeeeeeeelllll, that's the problem, I just can't underline it whole, because those are considered two different nodes 😅

Reply:
But what I can do is in this error message tell the user to 'Use build(:foo) instead'

Reply:
Well, when I run ruby-parse locally:
```
ruby-parse -e 'foo { association :foo }'
```
I get
```
(block
  (send nil :foo)
  (args)
  (send nil :association
    (sym :foo)))
```

If I remember correctly, you don't want to use `#on_send` in your cop, because your aim is getting the cop to be called on a block (first ast statement), hence the `#on_block` method 🤔

Reply:
Hmmmm, that could work, I'll check it out a bit. The problem that might arise is that the `block` part is completely optional

Reply:
yup, that might be it!

<img width="760" alt="Screenshot 2022-02-16 at 16 01 48" src="https://user-images.githubusercontent.com/4703945/154292350-f8bceee5-d1cc-491d-b0e4-a5f237c3c6de.png">

Reply:
Cool, will do the solution then so that it calls the on block part and matches it against one node_matcher and does the on_send part and matches it against the other one.

Reply:
So, after a lot of fighting with this. I finally managed to do it using the `on_block`. The problem with using the `on_block` was that it would still go into the `on_send` as correction gets memorised but not applied immediately, so I had to make sure that if the correction was applied that it should skip the `on_send` part. So, if there is still something that can be a bit improved or changed would love to as it looks much better now, but still a bit hacky